src/sync/mutex.rs

```diff
@@ -61,13 +61,16 @@ impl<T: ?Sized> Mutex<T> {
                 panic!("deadlock! task {:?} tried to acquire a Mutex it already holds", me);
             }
             ExecutionState::with(|s| s.current_mut().block());
+            drop(state);
+
+            // Note that we only need a context switch when we are blocked, but not if the lock is
+            // available. Consider that there is another thread `t` that also wants to acquire the
+            // lock. At the last context switch (where we were chosen), `t` must have been already
+            // runnable and could have been chosen by the scheduler instead.
```

Comment:
I think this is assuming that there is a context switch when a lock is dropped.  Because without a yield in the drop, a thread that releases a lock and then immediately tries to acquire it again will always succeed, which misses a potential interleaving.

Reply:
There *is* a context switch in drop (after doing the release):
https://github.com/bkragl/shuttle/blob/e47be262a3afe5590423f5e2b1b0927e698f4063/src/sync/mutex.rs#L266-L267

Reply:
Oh, I guess your point is that this should be part of the argument.

Reply:
Yes, was asking you to update the comment.

Reply:
Would be good to add a mutex unit test for the scenario Rajeev's describing (if we don't already have one somewhere).

Reply:
Updated the comment and added test `unlock_yields`.